src/main/java/com/petfinder/service/UserService.java

```diff
@@ -41,6 +46,63 @@ public void register(String login, String password, String repeatPassword,
 			}
 		}
 	}
+		
+	@Transactional
+	public void changeUserData(String currentPassword, String newPassword, String repeatPassword, String newEmail, String repeatEmail) throws PasswordsDoesNotMatchException, EmailExistsException, EmailsDoesNotMatchException, InvalidUserPasswordException{
+		if(checkIfUserIsLogged()){
+	        String login = SecurityContextHolder.getContext().getAuthentication().getName();
+			User user = userRepository.findOneByLogin(login);
+			if(this.passwordEncoder.matches(currentPassword, user.getPassword())){
+				boolean isPasswordUpdateRequired = passwordNeedsToBeUpdated(newPassword,repeatPassword);
+				boolean isEmailUpdateRequired = emailNeedsToBeUpdated(newEmail,repeatEmail);
+				
+				if(isPasswordUpdateRequired){
+					updatePassword(user,newPassword, repeatPassword);
+				}
+				if(isEmailUpdateRequired){
+					updateEmail(user,newEmail, repeatEmail);
+				}
+				
+				if(isPasswordUpdateRequired == true || isEmailUpdateRequired == true){
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid unnecessary comparisons in boolean expressions](https://www.codacy.com/app/matikz1110/PetFinder/file/2655555896/issues/source?bid=3239464&fileBranchId=3326962#l66)